### ISSUE TITLE
[React] Cannot find "toCanvas"

### ISSUE THREAD

**Issue by JKamsker** [metafloor/bwip-js#304]
I am using vanilla ``create-react-app`` with ``npm i bwip-js``. 

![grafik](https://github.com/metafloor/bwip-js/assets/11245306/3e9e6710-b218-4aed-8611-f7c15a187617)

my code is: 

```tsx
import React, { useEffect } from 'react';
import logo from './logo.svg';
import './App.css';
import bwipjs from 'bwip-js';

const App = () => {
  useEffect(() => {
    try {
      // The return value is the canvas element
      let canvas = bwipjs.toCanvas('mycanvas', {
        bcid:        'code128',       // Barcode type
        text:        '0123456789',    // Text to encode
        scale:       3,               // 3x scaling factor
        height:      10,              // Bar height, in millimeters
        includetext: true,            // Show human-readable text
        textxalign:  'center',        // Always good to set this
      });
    } catch (e) {
      // `e` may be a string or Error object
    }
  }, []); // Empty array means this effect runs once on mount and not on updates

  return (
    <div className="App">
      <div className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h2>Welcome to React</h2>
      </div>
      <canvas id="mycanvas"></canvas>
    </div>
  );
}

export default App;
```

package.json:
```json
{
  "name": "ktw-five",
  "version": "0.1.0",
  "keywords": [
    "react",
    "starter"
  ],
  "private": true,
  "dependencies": {
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@mui/material": "^5.13.7",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.5.2",
    "@types/node": "^16.18.38",
    "@types/react": "^18.2.14",
    "@types/react-dom": "^18.2.6",
    "bwip-js": "^4.0.1",
    "react": "^18.2.0",
    "react-async-script": "^1.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "react-webcam": "^7.1.1",
    "typescript": "^4.9.5",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```
npm -v : 8.9.0

the problem seems to be, that my app is detected as "node", rather than "browser". Is there any go-to solution for this?

**Comment by metafloor** [comment]
I just verified functionality by recreating my sample react app.  It works fine.  Checking the bundled source, I am seeing the following:

![image](https://github.com/metafloor/bwip-js/assets/385436/479fc7bc-255d-492b-89d1-05570a962f48)

Which is the browser version of bwip-js.  Is there a possibility that you have a global configuration for react that is causing it to bring in the node version.  (Note that I am _not_ a react developer, so no clue if that is even possible....)

**Comment by olereidar** [comment]
I have the exact same problem. The import seems to work... (I am getting the canvas working), but typescript is telling me that ` bwipjs.ToBufferOptions` and `bwipjs.toCanvas` does not exists.

The `bwipjs.toBuffer` seems to work with typescript.

**Comment by olereidar** [comment]
I found the problem. The bwip-js package has specified multiple entry points for different environments:

For browsers: "browser": "./dist/bwip-js.js"
For Node.js: "main": "./dist/bwip-js-node.js"
For React Native: "react-native": "./dist/bwip-js-rn.mjs"


The issue I'm facing seems to be related to the way my bundler (e.g., Webpack, Rollup, Vite) resolves the imports. When you're importing bwip-js in a browser environment, the bundler should ideally pick the file specified under the "browser" key. 

I simply fixed it by changing to `"moduleResolution": "Bundler",` in my tsconfig.json. 

Im using Vite btw
